### PR TITLE
Add use case documentation

### LINKED ISSUE #1: Add Use Case documentation
Add Use Case documentation to the docs, with example screenshots and requests/responses. Like https://checkr.github.io/flagr/#/flagr_use_cases

### PR DESCRIPTION
**Pull request by aaronraff**
Closes #1

This is still quite a bit of work to be done on this. I wanted to get @markphelps feedback on the A/B testing section before moving onwards.

Let me know what could be better! That way I can apply that sort of style throughout this doc.

### CODE REVIEW

File: content/docs/use_cases.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Use Cases
+
+Flipt provides a solution for **feature flagging**. The next few sections provide examples of how Flipt can be used.
+
+## A/B Testing
+
+A/B testing can be accomplished by setting up a percentage rollout in Flipt. First you will need to create a segment. This is detailed in [Getting Started]({{< relref "/docs/getting_started" >}}). You can simply create a segment with no constraints, which will match all users.
```
Comment (markphelps) [inline content/docs/use_cases.md]:
```suggestion
A/B testing can be accomplished by setting up a percentage rollout rule in Flipt. See the [Getting Started]({{< relref "/docs/getting_started" >}}) section for how accomplish this.
```

File: content/docs/use_cases.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+Next you will want to create a flag with some variants by going to **Flags -> New Flag**. For this example I have created two variants: group-a and group-b.
```
Comment (markphelps) [inline content/docs/use_cases.md]:
```suggestion
In this example rule, we have two variants that we are targeting, `group-a` and `group-b`, each with a 50% distribution.
```

File: content/docs/use_cases.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+After you have both your segment and flag created, head on over to the **Targeting** tab in the flag that you just made. Then select **New Rule**. From here you can match on your newly created segment and choose **A Percentage Rollout** from the dropdown.
```
Comment (markphelps) [inline content/docs/use_cases.md]:
```suggestion
```

File: content/docs/use_cases.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+Next, choose your percentages like so:
```
Comment (markphelps) [inline content/docs/use_cases.md]:
```suggestion
```

File: content/docs/use_cases.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+{{< figure src="/img/use_cases/flag_targeting_rollout.png" alt="Flag Targeting" >}}
+
+Now that your A/B testing flag is all set up you can begin to make requests! Using your favorite Flipt gRPC client library, you can call the `Evaluate` method with the request parameters.
```
Comment (markphelps) [inline content/docs/use_cases.md]:
```suggestion
An example request to evaluate this rule within your application would look like this:
```

File: content/docs/use_cases.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+		FlagKey:  "new-landing-page",
+		EntityId: "test@email.com",
+	})
+```
+
+{{< hint info >}}
+Notice that we are using an email address for the `EntityId`. This does not need to be an email address, it just needs to be a unique identifier for your users.
+{{< /hint >}}
+
+One of the fields in the reulting struct is `Value`. You can check this to determine what landing page to respond with:
```
Comment (markphelps) [inline content/docs/use_cases.md]:
```suggestion
The response will contain which variant we should show for this user, either `group-a` or `group-b`:
```

**Review by markphelps — CHANGES_REQUESTED** [review]
Thanks @aaronraff! I think this is a great start and really appreciate your help!

One thing I was thinking about for this use case documentation is that we could try and keep it high level, and focus on the 'what/why', instead of the 'how'. Much like they do with flagr here: https://checkr.github.io/flagr/#/flagr_use_cases, where they use psuedocode and don't really get into the how to create the flags, etc.

I think the Getting Started and Integration docs already do a pretty good job of walking the user how to set flags/segments/rules up via the UI.

I went ahead and made some suggestions, but don't take them as required changes.

Thank you again for all your help on Flipt and now the docs! ❤️

**Comment by aaronraff** [comment]
Thanks for the feedback @markphelps! Good call on the 'what/why' rather than 'how'. I'll clean this up a bit and add some more use cases in the next day or so.

**Review by markphelps — APPROVED** [review]
Looks great, thank you @aaronraff !

**Pull request merged**